From patches Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
To: Laszlo Ersek <lersek@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
 Roy Tam <roytam@gmail.com>, qemu-devel@nongnu.org,
 Peter Maydell <peter.maydell@linaro.org>
Cc: Laszlo Ersek <lersek@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
 Stefan Weil <sw@weilnetz.de>, qemu-devel@nongnu.org, qemu-stable@nongnu.org,
 jacob.kroon@gmail.com
Date: Wed, 20 Feb 2013 14:43:30 +0100
Message-id: 1361367811-13288-2-git-send-email-pbonzini@redhat.com
Subject: [PATCH 1/2] win32: do not set CPU affinity

QEMU system emulation has been thread-safe for a long time, and
setting the CPU affinity is hurting performance badly.  Remove
the bogus code.

Jacob Kroon reports that the time to boot his VxWorks image goes from
"3 minutes passed and I still haven't made it that far" to ~140s.

Cc: qemu-stable@nongnu.org
Tested-by: Jacob Kroon <jacob.kroon@gmail.com>
Reviewed-by: Laszlo Ersek <lersek@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
        Jacob, this is "patch 3" you tested.

 os-win32.c | 18 -------------------
 1 file changed, 18 deletions(-)

diff --git a/os-win32.c b/os-win32.c
index 9673a81..3d43604 100644
--- a/os-win32.c
+++ b/os-win32.c
@@ -69,25 +69,7 @@ static BOOL WINAPI qemu_ctrl_handler(DWORD type)
 
 void os_setup_early_signal_handling(void)
 {
-    /* Note: cpu_interrupt() is currently not SMP safe, so we force
-       QEMU to run on a single CPU */
-    HANDLE h;
-    DWORD_PTR mask, smask;
-    int i;
-
     SetConsoleCtrlHandler(qemu_ctrl_handler, TRUE);
-
-    h = GetCurrentProcess();
-    if (GetProcessAffinityMask(h, &mask, &smask)) {
-        for(i = 0; i < 32; i++) {
-            if (mask & (1 << i))
-                break;
-        }
-        if (i != 32) {
-            mask = 1 << i;
-            SetProcessAffinityMask(h, mask);
-        }
-    }
 }
 
 /* Look for support files in the same directory as the executable.  */
-- 
1.8.1.2



